src/global/header.ts

```diff
@@ -17,44 +19,113 @@ export async function initHeader (store: IndexedFormula, fetcher: Fetcher) {
+
   const banner = document.createElement("div")
   banner.classList.add("header-banner")
   banner.appendChild(podLink)
+  banner.appendChild(userMenu)
 
   return banner
 }
 
-function createProfileLink (store: IndexedFormula, profile: NamedNode): HTMLElement {
+function createUserMenu (store: IndexedFormula, profile: NamedNode): HTMLElement {
+  const profileImg = getProfileImg(store, profile)
+
+  const menuProfileLink = document.createElement("a")
+  menuProfileLink.classList.add("header-user-menu__link")
+  menuProfileLink.href = profile.uri
+  menuProfileLink.innerText = "Profile"
+
+  const menuProfileItem = document.createElement("li")
+  menuProfileItem.classList.add("header-user-menu__list-item")
+  menuProfileItem.appendChild(menuProfileLink)
+
+  const menuLogoutButton = document.createElement("button")
+  menuLogoutButton.classList.add("header-user-menu__button")
+  menuLogoutButton.type = "button"
+  menuLogoutButton.addEventListener("click", () => panes.UI.authn.solidAuthClient.logout())
+  menuLogoutButton.innerText = "Log out"
+
+  const menuLogoutItem = document.createElement("li")
+  menuLogoutItem.classList.add("header-user-menu__list-item")
+  menuLogoutItem.appendChild(menuLogoutButton)
+
+  const loggedInMenuList = document.createElement("ul")
+  loggedInMenuList.appendChild(menuProfileItem)
+  loggedInMenuList.appendChild(menuLogoutItem)
+
+  const loggedInMenu = document.createElement("nav")
+  loggedInMenu.classList.add("header-user-menu__navigation-menu")
+  loggedInMenu.setAttribute("aria-hidden", "true")
+  loggedInMenu.appendChild(loggedInMenuList)
+
+  const loggedInMenuTrigger = document.createElement("button")
+  loggedInMenuTrigger.classList.add("header-user-menu__trigger")
+  loggedInMenuTrigger.type = "button"
+  if (typeof profileImg === "string") {
+    loggedInMenuTrigger.innerHTML = profileImg
+  } else {
+    loggedInMenuTrigger.appendChild(profileImg)
+  }
+
+  const loggedInMenuContainer = document.createElement("div")
+  loggedInMenuContainer.classList.add("header-banner__user-menu", "header-user-menu")
+  loggedInMenuContainer.appendChild(loggedInMenuTrigger)
+  loggedInMenuContainer.appendChild(loggedInMenu)
+
+  const throttledMenuToggle = throttle((event: Event) => toggleMenu(event, loggedInMenuTrigger, loggedInMenu), 50)
+  loggedInMenuTrigger.addEventListener("click", throttledMenuToggle)
+  loggedInMenuContainer.addEventListener("mouseover", throttledMenuToggle)
+  loggedInMenuContainer.addEventListener("mouseout", throttledMenuToggle)
+
+  return loggedInMenuContainer
+}
+
+function createSubBanner (store: IndexedFormula, profile: NamedNode | null, powOwnerProfile: NamedNode): HTMLElement {
   const profileLinkPre = document.createElement("span")
   profileLinkPre.innerText = "You're visiting the Pod controlled by "
 
   const profileLink = document.createElement("a")
-  profileLink.href = profile.uri
+  profileLink.href = powOwnerProfile.uri
```

Comment:
I think this should theoretically use `OutlineManger.GotoSubject` or something like that? It works now, but does trigger a full page refresh.